config/base/src/lib.rs

```diff
@@ -256,3 +271,59 @@ pub mod view {
         pub const IS_HAS_VIEW: bool = true;
     }
 }
+
+pub mod proxy {
+    //! Module for configuration proxies' traits
+
+    use super::*;
+
+    /// Pseudo-default trait only used for doc generation
+    pub trait DocsDefault {
+        fn default() -> Self;
+    }
+
+    /// Trait used to convert configs from file and env
+    pub trait Combine: Sized + Serialize + DeserializeOwned {
+        /// Which type of [`Configuration`] it builds into
+        type Target;
+
+        /// Build the config, do the necessary checks
+        fn build(self) -> Result<Self::Target, derive::Error>;
+
+        /// If any of the fields in [`other`] are filled, they
+        /// override the values of the fields in [`self`].
+        fn combine(self, other: Self) -> eyre::Result<Self, eyre::Error>;
```

Comment:
Should return `iroha_config::base::proxy::Error`, which enumerates all possible failure modes. Eyre is only useful for quick and dirty front-end stuff, not buried deep inside a lbrary.